src/common.hh

```diff
@@ -59,16 +59,20 @@ struct site {
     double lat;
     double lon;
     float alt;
+    /* TODO: remove the following fields. They use a huge amount of
+       stack and conflate IO with baseness logic. */
     char name[50];
     char filename[255];
 };
 
-struct path {
+struct Path {
     std::vector<double> lat;
     std::vector<double> lon;
     std::vector<double> elevation;
     std::vector<double> distance;
-    int length;
+    ssize_t ssize() const;
+    Path() = default;
+    Path(site const &src, site const &dst);
```

Comment:
I don't understand this C++ stuff, what's happening here

Reply:
oh, it's a constructor!

Reply:
correct. It's so the Path is "correct by construction".